lib/search-suggestions/index.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import SmallSearchIcon from '../icons/search-small';
+import TagIcon from '../icons/tag';
+
+export class SearchSuggestions extends Component {
+  static displayName = 'SearchSuggestions';
+
+  static propTypes = {
+    onSearch: PropTypes.func.isRequired,
+    query: PropTypes.string.isRequired,
+    filteredTags: PropTypes.array.isRequired,
+    storeKeyHandler: PropTypes.func.isRequired,
+  };
+
+  state = {
+    selectedItem: 0,
+  };
+
+  componentDidMount() {
+    this.props.storeKeyHandler({
+      next: this.nextItem,
+      prev: this.prevItem,
+      select: this.selectItem,
+    });
+  }
+
+  componentDidUpdate() {
+    // scroll selected item into view
+    var selectedItem = this.refs.selectedItem;
+    var domNode = null;
+    // todo scroll only when needed (i.e. item is not visible)
+    if (selectedItem) {
+      domNode = ReactDOM.findDOMNode(selectedItem);
+      domNode.scrollIntoView(false);
+    }
+
+
+    this.props.storeKeyHandler({
+      next: this.nextItem,
+      prev: this.prevItem,
+      select: this.selectItem,
+    });
```

Comment:
if we extract the selected item from this component then I think we can probably get rid of all of the `prev/next/select` passing. 🤔 

to do that I think we'd have to just pass `onSelectionChanged` to circle back up update when we've passed invalid ranges (the index clamping) though we can eliminate the possibility of an invalid index by obliterating the index itself…

```js
// search-field
this.state = {
	activeSelection: null,
}

…

const { activeSelection } = this.state;
const { filteredTags } = this.props;
case KEY_UP:
	const prevSelection = Math.max( 0, filteredTags.indexOf( activeSelection ) - 1 );
	return this.setState( { activeSelection: filteredTags[ prevSelection ] } );

case KEY_DOWN:
	const nextSelection = Math.min( filteredTags.length - 1, filteredTags.indexOf( activeSelection ) + 1 );
	return this.setState( { activeSelection: filteredTags[ nextSelection ] } );

…

const query = …
const filteredTags = …

<SearchSuggestions query={ query } tags={ filteredTags } activeSelection={ this.state.activeSelection } />

…

// search-suggestions
const { query, tags, activeSelection } = this.props;

…

className={ activeSelection === query ? '…' : '…' }

…

ref={ activeSelection === tag ? this.selectedItem : null }
```

This involves some more head-work but also unravels other headwork. It's inherently safe because in the event that send an active selection which has somehow disappeared then we're not messing with indices and mismatches.

A change in behavior is that if our search results change the either the active selection will follow the previously-selected tag around with the results _or_ if that previously-selected tag disappears then so will our active selection. I don't know if this is better or worse than keeping the _position_ of the selection constant. I think there are advantages both ways.

What it would do though is remove some back-and-forth with prop passing. It looks like you have boiled this down to a pretty minimal set of exchanges so I wouldn't worry too much about it, though I think if we could do the refactor (maybe after this PR lands unless you are willing to try now) then the entire combination of the two components and their logic might become clearer and easier to reason about - we won't have to couple the parent and children together so tightly.